tools/db_bench_tool.cc

```diff
@@ -7353,33 +7356,21 @@ int db_bench_tool(int argc, char** argv) {
 
 #ifndef ROCKSDB_LITE
   // Check for the encryption flag (and create an encrypted environment)
-  Random rnd(301);
   std::shared_ptr<EncryptionProvider> provider;
-  if (FLAGS_encrypt_db && !strcasecmp(FLAGS_block_cipher.c_str(), "IPP_AES")) {
-    std::string key(FLAGS_block_cipher_key_size_in_bytes, ' ');
-    for (int i = 0; i < FLAGS_block_cipher_key_size_in_bytes; ++i)
-      key[i] = rnd.Uniform(256) & 0xFF;
-
-    std::string provider_str = "IPP_AES";
-    switch (FLAGS_block_cipher_key_size_in_bytes) {
-      case 16:
-        provider_str += ":128";
-        break;
-      case 24:
-        provider_str += ":192";
-        break;
-      case 32:
-        provider_str += ":256";
-        break;
-      default:
-        fprintf(stderr,
-                "key size for the AES block cipher must be 16, 24, or "
-                "32 bytes.\n");
-        exit(1);
+  if (FLAGS_encrypt_db &&
+      !strcmp(FLAGS_block_cipher.c_str(),
+              EncryptionProviderNames::kIPPAESProviderName)) {
+    if (FLAGS_block_cipher_key.size() != FLAGS_block_cipher_key_size_in_bytes) {
+      fprintf(stderr,
+              "The size of key doesn't match with "
+              "block_cipher_key_size_in_bytes.\n");
+      exit(1);
```

Comment:
Is this already done inside of the EncryptionProvider?  I wonder if you need it here or not?